diff --git a/arch/arm64/configs/lineageos_joan_defconfig b/arch/arm64/configs/lineageos_joan_defconfig
index 966fb653..c27aa690 100644
--- a/arch/arm64/configs/lineageos_joan_defconfig
+++ b/arch/arm64/configs/lineageos_joan_defconfig
@@ -727,3 +727,8 @@ CONFIG_CRYPTO_AES_ARM64_CE_BLK=y
 CONFIG_CRYPTO_AES_ARM64_NEON_BLK=y
 CONFIG_CRYPTO_CRC32_ARM64=y
 CONFIG_QMI_ENCDEC=y
+
+# KernelSU
+CONFIG_KSU=y
+CONFIG_KSU_MANUAL_HOOK=y
+# CONFIG_KSU_KPROBES_HOOK is not set
+CONFIG_KPROBES=y
+CONFIG_KPROBE_EVENTS=y
diff --git a/drivers/staging/qca-wifi-host-cmn/hif/src/ce/ce_main.c b/drivers/staging/qca-wifi-host-cmn/hif/src/ce/ce_main.c
index 2f7bdecf..42b01567 100644
--- a/drivers/staging/qca-wifi-host-cmn/hif/src/ce/ce_main.c
+++ b/drivers/staging/qca-wifi-host-cmn/hif/src/ce/ce_main.c
@@ -1750,7 +1750,7 @@ hif_send_head(struct hif_opaque_softc *hif_ctx,

 	if (qdf_unlikely(ce_hdl == NULL)) {
 		HIF_ERROR("%s: error CE handle is null", __func__);
-		return A_ERROR;
+		return QDF_STATUS_E_FAILURE;
 	}

 	QDF_NBUF_UPDATE_TX_PKT_COUNT(nbuf, QDF_NBUF_TX_PKT_HIF);
diff --git a/drivers/staging/qcacld-3.0/Kbuild b/drivers/staging/qcacld-3.0/Kbuild
index 36419dec..71232608 100644
--- a/drivers/staging/qcacld-3.0/Kbuild
+++ b/drivers/staging/qcacld-3.0/Kbuild
@@ -1659,6 +1659,7 @@ ccflags-y += $(INCS)
 cppflags-y +=	-DANI_OS_TYPE_ANDROID=6 \
 		-Wall\
 		-Werror\
+		-Wno-error=implicit-enum-enum-cast \
 		-D__linux__

 cppflags-$(CONFIG_PTT_SOCK_SVC_ENABLE) += -DPTT_SOCK_SVC_ENABLE
diff --git a/drivers/staging/qcacld-3.0/core/dp/htt/htt_h2t.c b/drivers/staging/qcacld-3.0/core/dp/htt/htt_h2t.c
index b2ca2424..43cf3501 100644
--- a/drivers/staging/qcacld-3.0/core/dp/htt/htt_h2t.c
+++ b/drivers/staging/qcacld-3.0/core/dp/htt/htt_h2t.c
@@ -616,7 +616,7 @@ htt_h2t_rx_ring_cfg_msg_hl(struct htt_pdev_t *pdev)

 	pkt = htt_htc_pkt_alloc(pdev);
 	if (!pkt)
-		return A_ERROR; /* failure */
+		return QDF_STATUS_E_FAILURE; /* failure */

 	/*
 	 * show that this is not a tx frame download
@@ -632,7 +632,7 @@ htt_h2t_rx_ring_cfg_msg_hl(struct htt_pdev_t *pdev)
 		HTC_HEADER_LEN + HTC_HDR_ALIGNMENT_PADDING, 4, true);
 	if (!msg) {
 		htt_htc_pkt_free(pdev, pkt);
-		return A_ERROR; /* failure */
+		return QDF_STATUS_E_FAILURE; /* failure */
 	}
 	/*
 	 * Set the length of the message.